azurelinuxagent/ga/exthandlers.py

```diff
@@ -298,11 +298,28 @@ def run(self):
             self.log_report = True
 
             if self._extension_processing_allowed() and self._incarnation_changed(etag):
-                activity_id, correlation_id, gs_creation_time = self.get_goal_state_debug_metadata()
-
-                logger.info(
-                    "ProcessGoalState started [Incarnation: {0}; Activity Id: {1}; Correlation Id: {2}; GS Creation Time: {3}]".format(
-                        etag, activity_id, correlation_id, gs_creation_time))
+                try:
+                    activity_id, correlation_id, gs_creation_time = self.get_goal_state_debug_metadata()
```

Comment:
A small suggestion, try parsing the data and if unavailable then return `NA` for the ones that we were unable to parse. Maybe move the try-catch inside the `get_goal_state_debug_metadata` function to make this main loop cleaner?